BootloaderCorePkg/Stage2/Stage2Hob.c

```diff
@@ -299,16 +299,14 @@ BuildBaseInfoHob (
   }
 
   // Build serial port hob
-  SerialPortInfo = BuildGuidHob (&gLoaderSerialPortInfoGuid, sizeof (SERIAL_PORT_INFO));
+  SerialPortInfo = BuildGuidHob (&gUniversalPayloadSerialPortInfoGuid, sizeof (UNIVERSAL_PAYLOAD_SERIAL_PORT_INFO));
   if (SerialPortInfo != NULL) {
-    SerialPortInfo->Revision    = LOADER_SERIAL_PORT_INFO_REVISION;
-    SerialPortInfo->Type        = 1;
-    SerialPortInfo->BaseAddr64  = 0x3F8;
-    SerialPortInfo->BaseAddr    = 0x3F8;
-    SerialPortInfo->Baud        = 115200;
-    SerialPortInfo->RegWidth    = 1;
-    SerialPortInfo->InputHertz  = 1843200;
-    SerialPortInfo->UartPciAddr = 0;
+    SerialPortInfo->Header.Revision = UNIVERSAL_PAYLOAD_SERIAL_PORT_INFO_REVISION;
+    SerialPortInfo->Header.Length   = sizeof (UNIVERSAL_PAYLOAD_SERIAL_PORT_INFO);
+    SerialPortInfo->RegisterBase    = GetSerialPortBase ();
```

Comment:
@gdong1 the serialportbase might change after PIC enumeration. Suggest to build the hob in later stage. fyi

Reply:
Good catch. Previous I just tested it using legacy uart port. Let me update the PR.